src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -453,6 +569,29 @@ class HuiMapCard extends LitElement implements LovelaceCard {
     }
   }
 
+  private async _getHistory(): Promise<void> {
+    const entityIds = this._configEntities
+      ?.map((entity) => entity.entity)
+      .join(",");
+    const endTime = new Date();
+    const startTime = new Date();
+    startTime.setHours(endTime.getHours() - this._config!.hours_to_show!);
+
+    const stateHistory = await fetchRecent(
+      this.hass,
+      entityIds,
+      startTime,
+      endTime
+    );
+
+    if (stateHistory.length < 1) {
+      return;
+    }
+
+    this._history = stateHistory;
+    this._date = new Date();
```

Comment:
You need to set this before you are requesting data, otherwise if 5 state changes happen while you are fetching data, you're still fetching the history 5 times.